src/arrayMethodJoin.js

```diff
@@ -3,9 +3,28 @@
 /**
  * Implement method join
  */
-function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+function applyCustomJoin(input) {
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
```

Comment:
A good place to use [the nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)